jetty-client/src/test/java/org/eclipse/jetty/client/jmx/HttpClientJMXTest.java

```diff
@@ -0,0 +1,51 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2017 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.client.jmx;
+
+import java.lang.management.ManagementFactory;
+import java.util.Locale;
+import java.util.Set;
+
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.jmx.MBeanContainer;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class HttpClientJMXTest
+{
+    @Test
+    public void testHttpClientName() throws Exception
+    {
+        MBeanServer mbeanServer = ManagementFactory.getPlatformMBeanServer();
+        MBeanContainer mbeanContainer = new MBeanContainer(mbeanServer);
+        HttpClient httpClient = new HttpClient();
+        httpClient.addBean(mbeanContainer);
```

Comment:
If you wanted to use a specific name instead of the default, you'd have to set it before the call to addBean()  or the call to start()? Maybe should document that?

Reply:
Very good point. Turns out that needs to be done before the call to `addBean()` otherwise the default name is used. I'll update the test case and add a comment.

Reply:
@janbartel I addressed your concerns, please review.